### PR TITLE
Strided batched axpy

### PR DESCRIPTION
**Pull request by YvanMokwinski**
Add batched and strided batched version of axpy plus few modifications.

Summary of proposed changes:
-  batched version of axpy with tests
-  strided batched version of axpy with tests
-  refactored non batched axpy but no tests modifications.
-  added utility routines in clients/include/norm.hpp and clients/include/unit.hpp

### CODE REVIEW

File: clients/include/testing_axpy_batched.hpp [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+#include "unit.hpp"
+#include "utility.hpp"
+
+/* ============================================================================================ */
+template <typename T>
+void testing_axpy_batched_bad_arg(const Arguments& arg)
+{
+    rocblas_local_handle handle;
+
+    rocblas_int N = 100, incx = 1, incy = 1, batch_count = arg.batch_count;
```
Comment (TorreZuk) [inline clients/include/testing_axpy_batched.hpp]: Not to be addressed in this PR, but we should discuss if want to do auto code review.  e.g.
DCL04-C. Do not declare more than one variable per declaration

File: clients/include/testing_axpy_strided_batched.hpp [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        CHECK_HIP_ERROR(dy.transfer_from(hy));
+
+        gpu_time_used = get_time_us(); // in microseconds
+        for(int iter = 0; iter < number_hot_calls; iter++)
+        {
+            rocblas_axpy_strided_batched<T>(
+                handle, N, &h_alpha, dx, incx, stridex, dy, incy, stridey, batch_count);
+        }
+        gpu_time_used     = (get_time_us() - gpu_time_used) / number_hot_calls;
+        rocblas_gflops    = axpy_gflop_count<T>(N * batch_count) / gpu_time_used * 1e6 * 1;
```
Comment (TorreZuk) [inline clients/include/testing_axpy_strided_batched.hpp]: * 1

Comment (YvanMokwinski) [reply clients/include/testing_axpy_strided_batched.hpp]: ?

File: clients/gtest/blas1_gtest.yaml [hunk 3]
```diff
@@ -93,6 +141,26 @@ Tests:
     function:
       - axpy:  *half_single_precisions_complex_real
 
+  - name: blas1_batched_with_alpha
+    category: nightly
+    N: [ 4000000 ]
+    incx_incy: *incx_incy_range
+    alpha_beta: *alpha_beta_range
+    alphai_betai: *alphai_betai_range
+    batch_count: [ 2 ]
```
[the remark this answers is not in the document]
Comment (YvanMokwinski) [reply clients/gtest/blas1_gtest.yaml]: good point

File: clients/include/testing_axpy_strided_batched.hpp (hunk 2 above)
Comment (TorreZuk) [reply clients/include/testing_axpy_strided_batched.hpp]: You tell me what the *1 at the end of the line is supposed to indicate  :-)

Comment (YvanMokwinski) [reply clients/include/testing_axpy_strided_batched.hpp]: :)

**Pull request merged**